### PR TITLE
[#PAB-427] reporting period range endpoint

### PR DESCRIPTION
**Pull request by cyrusdobbs**
https://trello.com/c/qT7fuTec/427-be-endpoint-list-of-financial-periods-covered-by-the-data

### Change description
Adds the `/reporting-period-range` endpoint that returns the first reporting period start date and the last reporting period end date in the Submission table.

- [X] Unit tests and other appropriate tests added or updated
- [X] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### CODE REVIEW

File: openapi/api.yml [hunk 1]
```diff
@@ -182,3 +182,21 @@ paths:
...
               schema:
                 $ref: 'components.yml#/components/schemas/GeneralError'
+
+  /reporting-period-range:
+    get:
+      summary: Returns the start and end date reporting period.
+      operationId: core.controllers.get_filters.get_reporting_period_range
+      responses:
+        '200':
+          description: Returns the earliest reporting period start date and the latest reporting period end date
```
Comment (nsnape) [inline openapi/api.yml:192]: is it worth mentioning datetime format/spec?

File: openapi/components.yml [hunk 2]
```diff
@@ -83,7 +83,17 @@ components:
...
+
+    ReportingPeriodRangeData:
+      type: object
+      properties:
+        start_date:
+          type: string
+          format: date-time
+        end_date:
+          type: string
+          format: date-time
```
Comment (nsnape) [inline openapi/components.yml:96]: might be worth adding example so datetime format is clearly shown?

File: tests/controller_tests/test_filters.py [hunk 3]
```diff
@@ -76,3 +82,60 @@ def test_get_outcome_categories(seeded_app_ctx):
...
+    sub4 = Submission(
+        submission_id="4",
+        submission_date=datetime(2021, 6, 5),
+        reporting_period_start=datetime(2021, 6, 1),
+        reporting_period_end=datetime(2021, 6, 30),
+        reporting_round=2,
+    )
+    submissions = [sub1, sub2, sub3, sub4]
+    db.session.add_all(submissions)
+    db.session.flush()
```
Comment (nsnape) [inline tests/controller_tests/test_filters.py:130]: we actually don't need the db.session.flush():

This actually is to run cascade deletions (ie flush out all the unwanted records). I think there are a few of these, they are a legacy of some copy-and-paste test code. We should remove all these others in a separate PR I guess

**Review by nsnape — CHANGES_REQUESTED** [review]

File: openapi/api.yml (hunk 1 above)
Comment (cyrusdobbs) [reply openapi/api.yml:192]: good shout

File: openapi/components.yml (hunk 2 above)
Comment (cyrusdobbs) [reply openapi/components.yml:96]: will do

**Pull request merged**